code/game/machinery/recycle_vendor.dm

```diff
@@ -888,6 +974,8 @@
 					chargecard.worth -= buyprice
 					var/datum/transaction/T = new(buyprice, chargecard.owner_name, "Material Purchase", src)
 					T.apply_to(my_account)
+			taxdebt += buyprice/32
+			settledebt()
```

Comment:
When the Silo conducts sales, it also has a 1/32 tax, and as a single private node that presumably doesn't get interacted with as often as the four existing public nodes, I decided to make it call `settledebt()` immediately upon doing so as well.